.github/workflows/release.yml

```diff
@@ -1,28 +1,126 @@
 name: Release
 
 on:
-  push:
-    tags:
-      - 'v[0-9]+.[0-9]+.[0-9]+'
+  workflow_dispatch:
+    inputs:
+      releaseType:
+        type: choice
+        options:
+          - minor
+          - patch
+        default: minor
+        required: true
+        description: 'minor: v0.X.0, patch: v0.0.X'
+      debug:
+        type: boolean
+        default: true
+        description: 'executes the workflow in debug mode (skip the publishing tag, docker image and release steps)'
 
 jobs:
-  release:
-    name: Create Release
-    if: github.event.base_ref == 'refs/heads/master'
-    runs-on: ubuntu-20.04
+  check-permission:
+    name: Check permission
+    if: contains(github.ref, 'refs/heads/master')
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
     steps:
-      - name: Get Version
-        id: get_version
+      - name: Check user permission
+        uses: actions-cool/check-user-permission@v2.2.0
+        id: check
+        with:
+          require: 'write'
+    outputs:
+      hasWritePermission: ${{ steps.check.outputs.require-result }}
+
+  publish-tag:
+    name: Publish tag
+    needs: check-permission
+    if: contains(needs.check-permission.outputs.hasWritePermission, 'true')
+    permissions:
+      contents: write
```

Comment:
Since we have already checked the permissions above, Is there a need for this?

Reply:
```
    if: contains(needs.check-permission.outputs.hasWritePermission, 'true')
```
Above line is related to user who runs release workflow. Its checks if user is admin and has write permission.

```
 permissions:
      contents: write
```
Github actions creates [GITHUB_TOKEN](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#about-the-github_token-secret).  This token is used to execute this yml step. Here `contents: write
` implies that create `GITHUB_TOKEN` with `contents: write` scope.